tests/test_sdsc.py

```diff
@@ -59,6 +59,14 @@ def test_highlighter():
     assert xml == "highlight <highlight>these two</highlight> words"
 
 
+def test_contextpatternlocations():
+    """checks the contextpatternlocations function"""
+    assert sdsc.contextpatternlocations(1, [1]) == [1]
+    assert sdsc.contextpatternlocations(1, [1, 2]) == [1, 2]
```

Comment:
Oh. factor should only ever be `1` (location=after), `-1` (location=before) or `1, -1` (location=both). I guess the function should validate this somehow.